librz/core/cmd_eval.c

```diff
@@ -237,29 +237,25 @@ RZ_IPI void rz_core_theme_nextpal(RzCore *core, int mode) {
 					}
 					eprintf("%s %s %s\n", rz_str_get_null(nfn), curtheme, fn);
 					if (nfn && !strcmp(nfn, curtheme)) {
-						rz_list_free(files);
-						files = NULL;
 						free(curtheme);
 						curtheme = strdup(fn);
 						RZ_FREE(home);
 						goto done;
 					}
 				} else {
 					if (!nextpal_item(core, mode, fn, ctr++)) {
-						rz_list_free(files);
-						files = NULL;
 						RZ_FREE(home);
 						goto done;
 					}
 				}
 			}
 		}
-		rz_list_free(files);
 		RZ_FREE(home);
 	}
 
 	path = rz_str_rz_prefix(RZ_THEMES RZ_SYS_DIR);
 	if (path) {
+		rz_list_free(files);
```

Comment:
why putting this here? can't we just free `files` before the `if`? That way we are sure it will be always freed, without having to think whether `path != NULL` also implies `home != NULL`, etc. etc.

Reply:
Make sure to use `rz_list_free` + `= NULL`